test/integration/federation/crud_test.go

```diff
@@ -65,20 +66,40 @@ func TestFederationCRUD(t *testing.T) {
 	}
 
 	t.Run("Resource should be propagated to a newly added cluster", func(t *testing.T) {
-		fixture, crudTester, obj := initCRUDTest(t, &fedFixture, adapterFactory, kind)
+		fixture, crudTester, obj, _ := initCRUDTest(t, &fedFixture, adapterFactory, kind)
 		defer fixture.TearDown(t)
 
 		updatedObj := crudTester.CheckCreate(obj)
 		// Start a new cluster and validate that the resource is propagated to it.
 		fedFixture.StartCluster(t)
 		// Check propagation to the new cluster by providing the updated set of clients
-		crudTester.CheckPropagationForClients(updatedObj, fedFixture.ClusterClients)
+		objectExpected := true
+		crudTester.CheckPropagationForClients(updatedObj, fedFixture.ClusterClients, objectExpected)
+	})
+
+	t.Run("Resource should only be propagated to the cluster with a matching selector", func(t *testing.T) {
+		fixture, crudTester, obj, adapter := initCRUDTest(t, &fedFixture, adapterFactory, kind)
+		defer fixture.TearDown(t)
+
+		// Set an annotation to specify that the object is isolated to cluster 1.
+		federatedtypes.SetAnnotation(adapter, obj, federationapi.FederationClusterSelectorAnnotation, `[{"key": "cluster", "operator": "==", "values": ["1"]}]`)
+
+		updatedObj := crudTester.Create(obj)
+
+		// Check propagation to the first cluster
+		objectExpected := true
+		crudTester.CheckPropagationForClients(updatedObj, fedFixture.ClusterClients[0:1], objectExpected)
+
+		// Verify the object is not sent to the second cluster
+		objectExpected = false
+		crudTester.CheckPropagationForClients(updatedObj, fedFixture.ClusterClients[1:2], objectExpected)
```

Comment:
Are you planning to add another test to verify that the resource is deleted from cluster 1 if cluster selector is updated to "cluster==2" after it was already propagated to cluster 1.